backend/remote-state/azure/client.go

```diff
@@ -39,7 +39,7 @@ func (c *RemoteClient) Get() (*remote.Payload, error) {
 	ctx := context.TODO()
 	blob, err := c.giovanniBlobClient.Get(ctx, c.accountName, c.containerName, c.keyName, options)
 	if err != nil {
-		if blob.Response.StatusCode == 404 {
```

Comment:
I think the nil pointer dereference comes from either `blob` being nil, or `blob.Response` being nil while trying to access `blob.Response.StatusCode`.
I think the fix should be:
`if blob != nil && blob.Response != nil && blob.Response.StatusCode == 404 {`

Reply:
From a higher level point of view, why is the error not being returned for a 404?
```
if err != nil {
    return nil, err
}
```

I would be surprised to see blob be populated when err is present.

Reply:
Hi, curtismj1, thanks for review

> I think the nil pointer dereference comes from either `blob` being nil, or `blob.Response` being nil while trying to access `blob.Response.StatusCode`.
> I think the fix should be:
> `if blob.Response != nil && blob.Response.StatusCode == 404 {`

`blob` and `blob.Response` not a pointer types. 

 


> From a higher level point of view, why is the error not being returned for a 404?
> 
> ```
> if err != nil {
>     return nil, err
> }
> ```
> 
> I would be surprised to see blob be populated with non-zero values when err is present.

At higher level it has some special handling for such case: https://github.com/hashicorp/terraform/blob/8057f19bb525dac68c8b033f2acfd6c7bace45d5/states/remote/state.go#L111-L122 
So i've just tried to not affect more pieces of code than necessary. Just a little fix for an annoying bug.